src/charm.py

```diff
@@ -1605,10 +1720,11 @@ def _pull_licenses(container: Container) -> None:
 
         for license_name in licenses:
             try:
-                license_file = container.pull(path=Config.get_license_path(license_name))
-                f = open(f"LICENSE_{license_name}", "x")
-                f.write(str(license_file.read()))
-                f.close()
+                # Lazy copy, only if the file wasn't already pulled.
+                filename = Path(f"LICENSE_{license_name}")
+                if not filename.is_file():
+                    license_file = container.pull(path=Config.get_license_path(license_name))
+                    filename.write_text(str(license_file.read()))
```

Comment:
newb q, is it necessary to close the file?

Reply:
[Pathlib's write_text](https://docs.python.org/3/library/pathlib.html#pathlib.Path.write_text) does the magic for us :)